pkg/kube/exec.go

```diff
@@ -113,32 +120,20 @@ func ExecOutput(cli kubernetes.Interface, namespace, pod, container string, comm
 		},
 	}
 
-	_, _, err := ExecWithOptions(cli, opts)
-	return err
+	return ExecWithOptions(cli, opts)
 }
 
-// ExecWithOptions executes a command in the specified container,
-// returning stdout, stderr and error. `options` allowed for
-// additional parameters to be passed.
-func ExecWithOptions(kubeCli kubernetes.Interface, options ExecOptions) (string, string, error) {
+// ExecWithOptions executes a command in the specified container, returning an error.
+// `options` allowed for additional parameters to be passed.
+func ExecWithOptions(kubeCli kubernetes.Interface, options ExecOptions) error {
```

Comment:
After looking at the desc, I thought, are there use cases where some consumers would need `Exec` and others would need `ExecWithOptions`, can we not just expose one. So that it's easier to use these APIs?

Reply:
LGTM otherwise.

Reply:
In the ideal case, of course, it would be better to use `ExecWithOptions` only. But, unfortunately, both are widely used.

Looking on the code, I figured out that `ExecOutput` is probably even more redundant.

I can do bigger refactoring and reduce number of execs, but it will be bigger change, so maybe we can do that by two or three steps ? This one, drops buffering in ExecWithOptions, Second one will remove `ExecOutput` and, if we will agree, that we'd like to do that, the third (and the biggest one) can get rid of pure `Exec` ?

Reply:
Now, that I think about it again, I don't think we should block the PR in this discussion. I think names are sufficient enough to not cause any confusion.
If we want to exec with some options use `ExecWithOptions` otherwise use `Exec`.